src/stores/messages/MessagesDataProviderStore.ts

```diff
@@ -387,4 +452,31 @@ export default class MessagesDataProviderStore {
 
 		return message;
 	};
+
+	@action
+	keepLoading = async (direction: 'next' | 'previous') => {
+		if (
+			this.messagesStore.filterStore.filter.streams.length === 0 ||
+			!this.searchChannelNext ||
+			!this.searchChannelPrev
+		)
+			return;
+
+		const query = {
+			...this.messagesStore.filterStore.messsagesSSEConfig.queryParams,
+			startTimestamp:
+				direction === 'previous' ? this.prevLoadEndTimestamp : this.nextLoadEndTimestamp,
+			searchDirection: direction,
+		} as MessagesSSEParams;
+
+		if (direction === 'previous') {
+			this.noMatchingMessagesPrev = false;
+			this.startPreviousMessagesChannel(query);
```

Comment:
You have optional argument interval, so if it wasnt passed keep_alive callback isnt passed to SSEChannel, is that intentional? we will never restrict loading range after user clicks resume?

Reply:
Yes